### ISSUE TITLE
Rule proposal: `prefer-spread` should handle some cases of `for..of`

### ISSUE THREAD

**Issue by Pyrolistical** [sindresorhus/eslint-plugin-unicorn#1713]
### Description

Often [`no-for-loop`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-for-loop.md) results in something that is a simple spread, which [`prefer-spread`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md) should handle

This works for both arrays and iterators.

### Fail

```js
const result = []
for (const child of expected) {
  result.push(child)
}
```

```js
const result = []
for (const [index, child] of expected.entries()) {
  result[index] = child
}
```

### Pass

```js
const result = [...array]
```

**Comment by sindresorhus** [comment]
Have you actually seen code like this in the wild?

**Comment by Pyrolistical** [comment]
@sindresorhus I'm embarrassed to say, yes, but in my own code. I don't set out to write a trivial spread, but often as I clean up for loops and delete code, it ends up being a trivial spread.

**Comment by sindresorhus** [comment]
Accepted
